### ISSUE TITLE
New default page only work upon previously closed app

### ISSUE THREAD

**Issue by th23x** [vicolo-dev/chrono#243]
**Describe the bug**
Great to have to option for a default page eg "Alarms" to be opened when launching Chrono - thanks for adding 👍

The function works as expexted when the app is previously closed - not open anymore in the recent apps overview. But it doesn't bring/show the set default page when the app was still open "in the background". 

**To Reproduce**
1. Set "alarms" as default page
2. Navigate to "timers" page
3. Move to Android start screen - ie Chrono gets hidden but runs in the background 
4. Click the app icon or (new) widget
5. Chrono comes the the foreground but shows the "timers" page, not the default "alarms" 

**Expected behavior**
Show default page once regaining focus

**Screenshots**
n/a

**Smartphone Information**
 - Device: Samsung S10e
 - OS: Android 12
 - App version 0.50-beta

**Additional context**

**Comment by AhsanSarwar45** [comment]
I don't think that would be desirable behavior. If you are doing something on one page, go to another app, and come back, you should continue where you left off rather than being suddenly taken back to default page.

But I can add another setting like 'Open default tab on focus' or something like that if someone does want this functionality.

**Comment by th23x** [comment]
Now that you mention I am not sure anymore...

Loosing / getting focus is potentially too much for forcing the default page again.

I stubled upon the idea when clicking on the new widget and expecting to get to the default.

**Comment by AhsanSarwar45** [comment]
Maybe a setting to open a specific page when launched from the widget. In the future when we might have stopwatch or timer widgets, they would ofcourse open the relevant pages when clicked. Maybe it might be desirable to control which page to open when clicking on clock widget too.

**Comment by th23x** [comment]
Yes, indeed... In case there are more widgets coming, potentially a setting in each widget "action on click" would be great with the options:

- open app (as is)
- open default page (as per general default setting)
- open specific page (alarms, timers,...)

Sounds like a good idea - and target user expectation as good as possible.

**Comment by th23x** [comment]
Interesting...usually not using the back button. 

But of course, if it ends the app, afterwards returns to the default tab. While hime button keeps the focus.

Maybe with many people and different desires for behavior best ist the option per widget. And add "last open" option the the default page settings.

**Comment by AhsanSarwar45** [comment]
> But app may be ended even when using home-button (OS kill)?

If you press the back button to close app, thats kills the app. I think its the standard way in android. So when you open it again, it starts the app and should send you to the default page.  

If you press the home button, that is supposed to minimize the app (send it to background). So when you open it again, it should send you where you left off. 

Ofcourse if the app is minimized for a long duration, the system kills it to save battery and resources. Opening it again will take you to the default page. 

All this is handled by the system and I don't think there is much to be controlled there. What we can control is the following 3 behaviors:

* What happens when app is cold started (current behavior is to open default page.  We can add an option to remember the last page and reopen that) 
* What happens when app is maximized from background (current behavior is to continue where you left off, and I think it should stay that way) 
* What happens after app opens by clicking on widget or notification (this should open the relevant user defined page), irrespective of whether the app was minimized or killed.
